toqito/matrix_props/is_permutation.py

```diff
@@ -0,0 +1,64 @@
+"""Is matrix a permutation matrix."""
+import numpy as np
+from toqito.matrix_props import is_square
+
+def is_permutation(mat: np.ndarray) -> bool:
+    r"""
+    Determine if a matrix is a permutation matrix [WikiPermutation]_.
+
+    A matrix is a permutation matrix if each row and column has a
+    single element of 1 and all others are 0.
+
+    Examples
+    ==========
+
+    Consider the following matrix
+
+    .. math::
+        A = \begin{pmatrix}
+                1 & 2 & 3 \\
+                4 & 5 & 6 \\
+                7 & 8 & 9
+            \end{pmatrix}
+
+    our function indicates that this is indeed a permutation matrix.
+
+    >>> from toqito.matrix_props import is_permutation
+    >>> import numpy as np
+    >>> A = np.array([[1, 0, 0], [0, 0, 1], [0, 1, 0]])
+    >>> is_permutation(A)
+    True
+
+    Alternatively, the following example matrix :math:`B` defined as
+
+    .. math::
+        B = \begin{pmatrix}

```

Comment:
The value of this matrix here does not align with the non-permutation matrix you have in the code example below.